client/src/components/messenger/Conversations.js

```diff
@@ -50,66 +52,76 @@ const AddConversationDialog = ({ open, onClose, token }) => {
                         />
                     </li>
-                )}
+                ))}
             </ul>
-            <button onClick={() => handleClose()} name="create" className="addConversation">Create</button>
+            <button onClick={() => handleClose()} name="create" className="addConversation">
+                Create
+            </button>
         </Dialog>
     );
 };
 
-
-const Conversations = ({ selectConversation }) => {
+const Conversations = ({ selectConversation, socket }) => {
     const [dialogOpen, setDialogOpen] = useState(false);
     const [conversations, setConversations] = useState([]);
-    const [init, setInit] = useState(false);
     const alertDialog = Alert.useAlertDialog();
 
     // Get the token from the users global state.
     const token = useSelector((state) => state.user.token);
+    //const { id } = useSelector((state) => state.user.userInfo);
+
+    const updateConversations = (newConversation) => {
+        setConversations((conversations) => {
+            return newConversation.length > 0 ? [...newConversation, ...conversations] : [...newConversation];
+        });
+    };
 
     useEffect(() => {
-        // if this is the first time rendering, get user conversations from server
-        if (!init) {
-            const runAsync = async () => setConversations(await ConversationsController.getConversations(token));
-            runAsync();
-            setInit(true);
-        }
-    }, [init]);
+        // send a request to the server to get conversations and setup socket to listen
+        // for conversations changes
+        ConversationsController.init(socket);
+        const runAsync = async () =>
+            setConversations(await ConversationsController.getConversations(token, updateConversations));
+        runAsync();
+    }, []);
 
     const handleClickAdd = () => setDialogOpen(true);
-    const addConversation = conversationDetails => {
+    const addConversation = (conversationDetails) => {
         setDialogOpen(false); // close add dialog
         // ask the server to create a new conversation with the list participants (and title)
         if (conversationDetails) {
             const runAsync = async () => {
                 let { title, list } = conversationDetails;
-                let result = await ConversationsController.createConversation(token, list, title);
-                if (result.success) { // add new conversation to the back of the list
-                    let { conversation } = result;
-                    setConversations(conversations.length > 0 ? [...conversations, conversation]: [conversation]);
-                    // auto open chat after creation
-                    selectConversation(conversation.id, conversation.title);
+                if (title.length < 1) {
+                    // create a title from the chosen participants' names
+                    let chop = list.length < 2 ? 20 : list.length < 3 ? 9 : 5;
```

Comment:
Maybe but these numbers in constants?

Reply:
This was probably a bad idea in the first place. For example, if the user didn't provide a title for the chat and the participant list was ["Melissa", "Krystal", "Christopher"], the title would end up being: "You, Melis, Kryst, Chris", which is not the best looking or could be confusing